app/src/main/java/com/github/polypoly/app/ui/game/TradeDialogsUIComponent.kt

```diff
@@ -0,0 +1,234 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.window.Dialog
+import com.github.polypoly.app.base.game.Player
+import com.github.polypoly.app.base.game.TradeRequest
+import com.github.polypoly.app.models.game.GameViewModel
+import com.github.polypoly.app.ui.theme.Padding
+import com.github.polypoly.app.ui.theme.Shapes
+
+/**
+ * Dialog to that propose a trade to the player
+ * @param trade The trade to propose
+ * @param openDialog A mutable state to open and close the dialog.
+ * @param gameModel The game view model
+ */
+@Composable
+fun ProposeTradeDialog(trade: TradeRequest, openDialog: MutableState<Boolean>, gameModel: GameViewModel) {
+    val openDialogChooseLocation = remember {
+        mutableStateOf(false)
+    }
+    Surface(
+        color = Color.Transparent,
+    ) {
+        if (openDialogChooseLocation.value) {
+            LocationsDialog(
+                "Choose a location to trade2",
+                openDialogChooseLocation,
+                trade.playerReceiver.getOwnedLocations()
+            ) {
+                trade.locationReceived = it
+                gameModel.updateTradeRequest(trade)
+                openDialogChooseLocation.value = false
+            }
+        }
+        if(openDialog.value) {
+            PropositionTradeDialog(trade, openDialog, gameModel, openDialogChooseLocation)
+        }
+    }
+}
+
+/**
+ * Dialog to know if the player accept the trade
+ * @param trade The trade to accept
+ * @param openDialog A mutable state to open and close the dialog.
+ * @param gameModel The game view model
+ * @param openDialogChooseLocation A mutable state to open and close the dialog to choose a location
+ */
+@Composable
+fun PropositionTradeDialog(trade: TradeRequest, openDialog: MutableState<Boolean>, gameModel: GameViewModel,
+                           openDialogChooseLocation: MutableState<Boolean>){
+    AlertDialog(
+        modifier = Modifier.testTag("propose_trade_dialog"),
+        onDismissRequest = { openDialog.value = false },
+        title = { Text(text = "Player ${trade.playerApplicant.user.name} propose you a trade !") },
+        text = { Text(text = "Do you accept ?") },
+        buttons = {
+            Row(
+                modifier = Modifier
+                    .padding(Padding.medium)
+                    .fillMaxWidth()
+            ) {
+                Button(
+                    onClick = {
+                        // test if you can accept the trade, i.e. if you have a location to trade
+                        if(trade.playerReceiver.getOwnedLocations().isNotEmpty()) {
+                            openDialogChooseLocation.value = true
+                        } else {
+                            trade.currentPlayerReceiverAcceptation = false
+                            gameModel.updateTradeRequest(trade)
```

Comment:
To simplify/modularize the code and move the business logic to the view model you could have instead functions such as `accept/rejectCurrentTradeRequest` in gameModel which could be reused when accepting the trade request as the applicant too. It would also be a better encapsulation to avoid having to set manually the new tradeRequest updated (see comment above regarding `updateTradeRequest`)

Reply:
done